src/R/Support/Impl/Help/Packages/PackageInfo.cs

```diff
@@ -2,257 +2,113 @@
-            List<INamedItemInfo> functions = new List<INamedItemInfo>();
-            string content = null;
-
-            try {
-                string htmlFile = Path.Combine(this.InstallPath, this.Name, "html", "00index.html");
-                if (File.Exists(htmlFile)) {
-                    using (StreamReader sr = new StreamReader(htmlFile, Encoding.UTF8)) {
-                        content = sr.ReadToEnd();
+                    using (var sw = new StreamWriter(filePath)) {
+                        foreach (var f in _functions) {
+                            sw.WriteLine(f.Name);
+                        }
                     }
+                    _saved = true;
+                } catch (IOException ioex) {
+                    GeneralLog.Write(ioex);
+                } catch (AccessViolationException aex) {
+                    GeneralLog.Write(aex);
                 }
-            } catch (IOException) { }
-
-            if (!string.IsNullOrEmpty(content)) {
-                HtmlTree tree = new HtmlTree(new TextStream(content));
-                tree.Build();
-
-                FunctionSearch functionSearch = new FunctionSearch(functions);
-                tree.Accept(functionSearch, null);
             }
+        }
+        #endregion
 
-            Dictionary<string, INamedItemInfo> functionIndex = new Dictionary<string, INamedItemInfo>();
-            foreach (INamedItemInfo ni in functions) {
-                functionIndex[ni.Name] = ni;
+        public async Task LoadFunctionsIndexAsync() {
+            var functionNames = await GetFunctionNamesAsync();
+            foreach (var functionName in functionNames) {
+                _functions.Add(new FunctionInfo(functionName));
             }
+        }
 
-            IReadOnlyDictionary<string, string> mappedNames = GetMappedNames();
-            foreach (string mappedName in mappedNames.Keys) {
-                INamedItemInfo ni;
-                string actualName = mappedNames[mappedName];
-                if (functionIndex.TryGetValue(actualName, out ni)) {
-                    INamedItemInfo niAlias = new NamedItemInfo() {
-                        Name = mappedName,
-                        Description = ni.Description,
-                        ItemType = ni.ItemType
-                    };
-                    functions.Add(niAlias);
-                }
+        private async Task<IEnumerable<string>> GetFunctionNamesAsync() {
+            var cached = TryRestoreFromCache();
+            if (cached == null) {
+                try {
+                    var r = await _host.Session.EvaluateAsync<JArray>(Invariant($"as.list(base::getNamespaceExports('{this.Name}'))"), REvaluationKind.Normal);
+                    return r.Select(p => (string)((JValue)p).Value).ToArray();
+                } catch (MessageTransportException) { } catch (TaskCanceledException) { } catch (REvaluationException) { }
+            } else {
+                _saved = true;
```

Comment:
Got it

---
In reply to: [71204403](https://github.com/Microsoft/RTVS/pull/2084#discussion_r71204403) [](ancestors = 71204403)